reference/conanfile/attributes.rst

```diff
@@ -1,6 +1,27 @@
 Attributes
 ==========
 
+name
+----
+This is a string, with a minimun of 2 and a maximum of 50 characters (though shorter names are recommended), that defines the package name. It will be the ``<PkgName>/version@user/channel`` of the package reference.
+It should match the following regex ``^[a-zA-Z0-9_][a-zA-Z0-9_\+\.-]$``, so start with alphanumeric or underscore, then alphanumeric, underscore, +, ., - characters.
+
+The name is only necessary for ``export``-ing the recipe into the local cache (``export`` and ``create`` commands), if they are not defined in the command line.
```

Comment:
I suspect that the name is also necessary for passing package-specific profiles settings, e.g.:

```sh
conan install \
  --build=missing \
  --env my-project-name:CXXFLAGS="-fsanitize=address,undefined -Wall -Werror" \
  ..
```